### PR TITLE
✨amp-brightcove: extend analytics support

### LINKED ISSUE #19817: Video analytics (currentTime, duration, playedRangesJson, and playedTotal) for amp-brightcove
Working with major UK publishers who use Brightcove and Permutive in a PWAMP set up and 

Some key metrics aren't available according to the docs here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-video-analytics.md#common-variables

In terms of 'partial support'  - Partial support means that all video analytics common variables are included, except for currentTime, duration, playedRangesJson, and playedTotal.

Is there a plan to introduce these common variables in amp-brightcove?

### PR DESCRIPTION
**Pull request by mister-ben**
- Implements currentTime, duration, playedRangesJson, and playedTotal in amp-brightcove
- Fixes #19817

### CODE REVIEW

**Comment by torch2424** [comment]
cc @aghassemi @alanorozco

File: extensions/amp-brightcove/0.1/amp-brightcove.js [hunk 1]
```diff
@@ -189,6 +198,16 @@ class AmpBrightcove extends AMP.BaseElement {
       this.playing_ = false;
     }
 
+    if (data.ct) {
```
Comment (cathyxz) [inline extensions/amp-brightcove/0.1/amp-brightcove.js]: It seems like closure isn't too happy with your dot access on `data`. That's why it is failing [Travis](https://travis-ci.org/ampproject/amphtml/jobs/471296017)
```
...Compiler issues for check-types.js:
extensions/amp-brightcove/0.1/amp-brightcove.js:201: ERROR - Cannot do '.' access on a dict
    if (data.ct) {
             ^^
extensions/amp-brightcove/0.1/amp-brightcove.js:202: ERROR - Cannot do '.' access on a dict
      this.currentTime_ = data.ct;
                               ^^
extensions/amp-brightcove/0.1/amp-brightcove.js:204: ERROR - Cannot do '.' access on a dict
    if (data.pr) {
             ^^
extensions/amp-brightcove/0.1/amp-brightcove.js:205: ERROR - Cannot do '.' access on a dict
      this.playedRanges_ = data.pr;
                                ^^
extensions/amp-brightcove/0.1/amp-brightcove.js:207: ERROR - Cannot do '.' access on a dict
    if (data.dur) {
             ^^^
extensions/amp-brightcove/0.1/amp-brightcove.js:208: ERROR - Cannot do '.' access on a dict
      this.duration_ = data.dur;
                            ^^^
```

You can run type check locally via `gulp check-types` to verify fix.

Comment (mister-ben) [reply extensions/amp-brightcove/0.1/amp-brightcove.js]: Got it. Changed to bracket notation.

**Comment by cathyxz** [comment]
@danielrozenberg do you know why this PR increases the bundle-size? `<amp-brightcove>` is an extension and definitely not part of the runtime. (FYI @mister-ben this check is non-blocking.) 
LGTM, but deferring to video components owner @alanorozco for approval and merge.

**Comment by aghassemi** [comment]
Thanks @mister-ben! Merged.

**Comment by danielrozenberg** [comment]
@cathyxz, regarding the bundle size, I'm still working out some of the kinks in the check (there's a reason why it's still non-blocking), but these issues mostly stem from how we select the base commit that the bundle size is compared against. @rsimha fixed that (hopefully forever) but some older PRs might still select the wrong baseline

**Pull request merged**